### PR TITLE
DOC: GH48089 Update doc for df to latex

### LINKED ISSUE #48089: DOC: Styler.to_latex link is broken in DataFrame.to_latex documentation
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_latex.html

### Documentation problem

In the See Also section of [DataFrame.to_latex](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_latex.html) documentation the element Styler.to_latex has broken internal reference link.

### Suggested fix for documentation

It could be fixed by adding the[ Styler.to_latex link](https://pandas.pydata.org/docs/reference/api/pandas.io.formats.style.Styler.to_latex.html) to the corresponding element in the See Also section. This would make the doc more complete.

### PR DESCRIPTION
**Pull request by samrao1997**
- [x] closes #48089  (Replace xxxx with the Github issue number)
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Review by datapythonista — APPROVED** [review]
Thanks for the fix and spending the time to research the problem @samrao1997, really nice.

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3351,7 +3351,8 @@ def to_latex(
         {returns}
         See Also
         --------
-        Styler.to_latex : Render a DataFrame to LaTeX with conditional formatting.
+        io.formats.style.Styler.to_latex : Render a DataFrame to LaTeX
```
Comment (phofl) [inline pandas/core/generic.py:3354]: Would creating a link work? Like

```
:meth:` ~io.formats.style.Styler.to_latex`
```

Comment (phofl) [reply pandas/core/generic.py:3354]: Forget it, was not aware that this is done automatically

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @samrao1997

**Pull request merged**